truss/tests/cli/test_model_team_resolver.py

```diff
@@ -0,0 +1,279 @@
+"""Tests for team parameter in model push.
+
+This test suite covers all 8 scenarios for team resolution in truss push:
+1. --team PROVIDED: Valid team name, user has access
+2. --team PROVIDED: Invalid team name (does not exist)
+3. --team NOT PROVIDED: User has multiple teams, no existing model
+4. --team NOT PROVIDED: User has multiple teams, existing model in exactly one team
+5. --team NOT PROVIDED: User has multiple teams, existing model exists in multiple teams
+6. --team NOT PROVIDED: User has exactly one team, no existing model
+7. --team NOT PROVIDED: User has exactly one team, existing model matches the team
+8. --team NOT PROVIDED: User has exactly one team, existing model exists in different team
+"""
+
+from unittest.mock import Mock, patch
+
+import click
+import pytest
+
+from truss.cli.resolvers.model_team_resolver import resolve_model_team_name
+from truss.remote.baseten.remote import BasetenRemote
+
+
+class TestModelTeamResolver:
+    """Test team parameter resolution for model push."""
+
+    @staticmethod
+    def _setup_mock_remote(teams):
+        mock_remote = Mock(spec=BasetenRemote)
+        mock_api = Mock()
+        mock_remote.api = mock_api
+        mock_api.get_teams.return_value = teams
+        return mock_remote
+
+    @pytest.mark.parametrize(
+        "provided_team_name,expected_team_name,expected_team_id,should_raise",
+        [
+            # SCENARIO 1: Valid team name
+            ("Team Alpha", "Team Alpha", "team1", False),
+            # SCENARIO 2: Invalid team name
+            ("NonExistentTeam", None, None, True),
+        ],
+    )
+    def test_team_provided_scenarios(
+        self, provided_team_name, expected_team_name, expected_team_id, should_raise
+    ):
+        """Test scenarios when --team is provided."""
+        teams = {"Team Alpha": {"id": "team1", "name": "Team Alpha"}}
+        mock_remote = self._setup_mock_remote(teams)
+
+        if should_raise:
+            with pytest.raises(click.ClickException) as exc_info:
+                resolve_model_team_name(
+                    remote_provider=mock_remote,
+                    provided_team_name=provided_team_name,
+                    existing_teams=teams,
+                )
+            assert "does not exist" in str(exc_info.value)
+            assert provided_team_name in str(exc_info.value)
+        else:
+            team_name, team_id = resolve_model_team_name(
+                remote_provider=mock_remote,
+                provided_team_name=provided_team_name,
+                existing_teams=teams,
```

Comment:
do we ever pass in existing_teams? i didn't see it passed in anywhere other than the tests